src/compiler/utilitiesPublic.ts

```diff
@@ -1241,6 +1241,31 @@ namespace ts {
         }
     }
 
+    /* @internal */
+    export function isThisOrSuperInStaticFieldBoundary(node: Node, parent: Node | undefined): boolean {

```

Comment:
This is used by a transformer. Transformer's shouldn't rely on `parent` pointers. Please see my note in *classFields.ts* about an alternative approach to this.